File: docs/snippets/python/docker/usage.py
```python
import subprocess
import httpx
import json
from pathlib import Path

class DockerKreuzbergClient:
    def __init__(self, container_name: str = "kreuzberg-api", port: int = 8000):
        self.container_name = container_name
        self.port = port
        self.api_url = f"http://localhost:{port}/api/extract"

    def start_container(self, image: str = "kreuzberg:latest"):
        print("Starting Kreuzberg Docker container...")
        subprocess.run(
            [
                "docker", "run", "-d",
                "--name", self.container_name,
                "-p", f"{self.port}:8000",
                image,
            ],
            check=True,
        )
        print(f"Container started on http://localhost:{self.port}")

    async def extract_file(self, file_path: str) -> str:
        file_bytes = Path(file_path).read_bytes()
        files = {"file": (Path(file_path).name, file_bytes)}

        async with httpx.AsyncClient() as client:
            response = await client.post(self.api_url, files=files)
            response.raise_for_status()
            result = response.json()
            return result.get("content", "")

    def stop_container(self):
        print("Stopping Kreuzberg Docker container...")
        subprocess.run(["docker", "stop", self.container_name], check=True)
        subprocess.run(["docker", "rm", self.container_name], check=True)
        print("Container stopped and removed")

async def main():
    docker_client = DockerKreuzbergClient()

    try:
        docker_client.start_container()
        import asyncio
        await asyncio.sleep(2)

        content = await docker_client.extract_file("document.pdf")
        print(f"Extracted content:\n{content}")
    finally:
        docker_client.stop_container()

if __name__ == "__main__":
    import asyncio
    asyncio.run(main())
```

File: docs/snippets/ruby/docker/usage.rb
```ruby
require 'open3'
require 'net/http'
require 'json'

class DockerKreuzbergClient
  def initialize(container_name = 'kreuzberg-api', api_port = 8000)
    @container_name = container_name
    @api_port = api_port
    @api_url = "http://localhost:#{api_port}/api/extract"
  end

  def start_container(image = 'kreuzberg:latest')
    puts 'Starting Kreuzberg Docker container...'
    cmd = "docker run -d --name #{@container_name} -p #{@api_port}:8000 #{image}"
    stdout, stderr, status = Open3.capture3(cmd)

    raise "Failed to start container: #{stderr}" unless status.success?

    puts "Container started on http://localhost:#{@api_port}"
  end

  def extract_file(file_path)
    file_content = File.read(file_path, mode: 'rb')
    boundary = "----WebKitFormBoundary#{SecureRandom.hex(16)}"

    body = "--#{boundary}\r\n"
    body += "Content-Disposition: form-data; name=\"file\"; filename=\"#{File.basename(file_path)}\"\r\n"
    body += "Content-Type: application/octet-stream\r\n\r\n"
    body += file_content
    body += "\r\n--#{boundary}--\r\n"

    uri = URI(@api_url)
    http = Net::HTTP.new(uri.host, uri.port)
    request = Net::HTTP::Post.new(uri.path)
    request['Content-Type'] = "multipart/form-data; boundary=#{boundary}"
    request.body = body

    response = http.request(request)
    result = JSON.parse(response.body)
    result['content']
  end

  def stop_container
    puts 'Stopping Kreuzberg Docker container...'
    system("docker stop #{@container_name}")
    system("docker rm #{@container_name}")
    puts 'Container stopped and removed'
  end
end

docker_client = DockerKreuzbergClient.new

begin
  docker_client.start_container
  sleep(2)

  content = docker_client.extract_file('document.pdf')
  puts "Extracted content:\n#{content}"
ensure
  docker_client.stop_container
end
```
